src/org/labkey/cds/data/CDSImportCopyConfig.java

```diff
@@ -33,37 +33,56 @@
 import org.labkey.api.query.QueryService;
 import org.labkey.api.query.QueryUpdateService;
 import org.labkey.api.query.ValidationException;
+import org.labkey.api.reader.DataLoaderFactory;
+import org.labkey.api.reader.DataLoaderService;
 import org.labkey.api.security.User;
 import org.labkey.api.util.FileUtil;
 
 import java.io.File;
 import java.io.IOException;
 import java.sql.ResultSet;
+import java.util.List;
 
-public abstract class CDSImportCopyConfig extends CopyConfig
+public class CDSImportCopyConfig extends CopyConfig
 {
-    public abstract String getFileExtension();
-    public abstract DataIteratorBuilder getTabLoader(File file) throws IOException;
-
     private final String _fileName;
+    private File _importFile;
+    private static final List<String> VALID_EXTENSIONS = List.of(".xls", ".xlsx", ".csv", ".tsv", ".txt");
+
     QueryUpdateService.InsertOption _option = QueryUpdateService.InsertOption.IMPORT;
 
     /**
-     *
-     * @param sourceSchema Describes whether the source is TSV or CSV delimited
-     * @param target The target table name
-     * @param targetSchema The target schema name
-     * @param fileName The filename containing the data to be imported
-     * @param copyToImportTable Copy into an intermediate table with an import_ prefix, this can be useful if further transformation
-     *                          needs to be performed before copying to the final destination. Otherwise, set false to
-     *                          copy directly into the destination table.
+     * Constructor which imports a text file into a table in the CDS schema.
+     * @param targetQuery the target table
+     * @param fileName the name of the file, without the extension
+     */
+    public CDSImportCopyConfig(String targetQuery, String fileName)
+    {
+        super(null, null, "cds", targetQuery);
```

Comment:
Since we're setting sourceSchema and sourceQuery to null via super(null, null...) here, on lines 91 and 97, any calls to this.getSourceSchema() and getSourceQuery() will return null. It seems that the code block from lines 89-100 may no longer be relevant due to your changes, but further testing is needed to confirm this.